File: app/src/main/java/com/bilgehandemirkaya/manbox/database/ActivityDB/ActivitySystemRepository.java
package com.bilgehandemirkaya.manbox.database.ActivityDB;

import androidx.lifecycle.LiveData



/*
Used to access multiple data sources. It is used to seperate code and the architecture
 */
class OperatingSystemRepository(private val ActivitySystemDAO: ActivitySystemDAO) {
    val readAlldata: LiveData<List<ActivitySystem>> = ActivitySystemDAO.getAllOperatingSystems()

    fun insertOperatingSystem(operatingSystem: OperatingSystems){
        operatingSystemDAO.insertOperatingSystem(operatingSystem)
    }
    fun insertOperatingSystems(operatingSystems:ArrayList<OperatingSystems>){
        operatingSystemDAO.insertAllOperatingSystems(operatingSystems)
    }

    fun updateOperatingSystem(operatingSystem: OperatingSystems){
        operatingSystemDAO.updateOperatingSystem(operatingSystem)
    }

    fun deleteOperatingSystem(operatingSystem: OperatingSystems){
        operatingSystemDAO.deleteOperatingSystem(operatingSystem)
    }

    fun deleteAllOperatingSystems(){
        operatingSystemDAO.deleteAllOperatingSystems()
    }

    fun getAllOperatingSystems(): LiveData<List<OperatingSystems>> {
        return operatingSystemDAO.getAllOperatingSystems()
    }

    fun getOperationsSystemsById(id:Int):OperatingSystems{
        return operatingSystemDAO.getOperatingSystemById(id)
    }
    fun getOperatingSystemsBySearchKey(searchKey:String): Flow<List<OperatingSystems>> {
        return operatingSystemDAO.getOperatingSystemsBySearchKey(searchKey)
    }
}

File: app/src/main/java/com/bilgehandemirkaya/manbox/database/ActivityDB/ActivitySystemRepository.kt
package com.bilgehandemirkaya.manbox.database.ActivityDB

import androidx.lifecycle.LiveData
import kotlinx.coroutines.flow.Flow


/*
Used to access multiple data sources. It is used to seperate code and the architecture
 */
class ActivitySystemRepository(private val activitySystemDAO: ActivitySystemDAO) {
    val readAlldata: LiveData<List<ActivitySystem>> = activitySystemDAO.getAllActivitySystems()

    fun insertActivitySystem(activitySystem: ActivitySystem){
        activitySystemDAO.insertActivitySystem(activitySystem)
    }


    fun updateActivitySystem(activitySystem: ActivitySystem){
        activitySystemDAO.updateActivitySystem(activitySystem)
    }

    fun deleteActivitySystem(activitySystem: ActivitySystem){
        activitySystemDAO.deleteActivitySystem(activitySystem)
    }

    fun deleteAllActivitySystems(){
        activitySystemDAO.deleteAllActivitySystems()
    }

    fun getAllActivitySystems(): LiveData<List<ActivitySystem>> {
        return activitySystemDAO.getAllActivitySystems()
    }



}
